src/main/java/com/synopsys/integration/blackduck/nexus3/task/metadata/MetaDataTask.java

```diff
@@ -144,18 +178,45 @@ protected void execute(final Repository repository) {
                 pagedAssets = commonRepositoryTaskHelper.retrievePagedAssets(foundRepository, nextPage);
             }
 
-            if (isProxyRepo && null != hubServicesFactory) {
+            if (isProxyRepo && null != blackDuckServicesFactory) {
                 logger.info("Updating data of proxy repository.");
                 try {
-                    final ProjectVersionWrapper projectVersionWrapper = inspectorMetaDataProcessor.getProjectVersionWrapper(hubServicesFactory, repoName);
-                    inspectorMetaDataProcessor.updateRepositoryMetaData(hubServicesFactory, projectVersionWrapper.getProjectVersionView(), assetWrapperMap, TaskStatus.SUCCESS);
+                    final Optional<ProjectVersionWrapper> projectVersionWrapperOptional = inspectorMetaDataProcessor.getProjectVersionWrapper(blackDuckServicesFactory, repoName);
```

Comment:
is there no way to pull this code all into a single common class? Looks like similar things in all 3 tasks. It seems relatively similar but it's tough to tell in the git diff.